### PR TITLE
More joypad models

### PR DESCRIPTION
**Pull request by Bytezz**
Added:
- DS3
- DualSense
- Xbox 360

Improved auto model recognition, following names at [SDL_GameControllerDB](https://github.com/gabomdq/SDL_GameControllerDB/blob/master/gamecontrollerdb.txt).

### CODE REVIEW

File: addons/ActionIcon/ActionIcon.gd [hunk 1]
```diff
@@ -9,7 +9,7 @@ const CUSTOM_ACTIONS = {
 enum {KEYBOARD, MOUSE, JOYPAD}
 enum JoypadMode {ADAPTIVE, FORCE_KEYBOARD, FORCE_JOYPAD}
 enum FitMode {NONE, MATCH_WIDTH, MATCH_HEIGHT}
-enum JoypadModel {AUTO, XBOX, DUALSHOCK, JOY_CON}
+enum JoypadModel {AUTO, Xbox, Xbox360, DS3, DS4, DualSense, Joy_Con}
```
Comment (KoBeWi) [inline addons/ActionIcon/ActionIcon.gd:12]: Constants should stay upper-case.

I see that you used a clever trick to remove ifs, but this forces certain naming on the files, which is meh. You could change the original name code to use `match` though.

File: addons/ActionIcon/ActionIcon.gd [hunk 2]
```diff
@@ -313,14 +313,19 @@ func get_joypad_model(device: int) -> String:
...
+			model = "DS3"
+		elif device_name.find("PS4") > -1:
+			model = "DS4"
+		elif device_name.find("PS5") > -1:
+			model = "DualSense"
+		elif device_name.find("Joy-Con") > -1 or device_name.find("Joy Con") > -1:
+			model = "Joy_Con"
+	else:
+		model = JoypadModel.keys()[joypad_model]
+		print(model)
```
Comment (KoBeWi) [inline addons/ActionIcon/ActionIcon.gd]: Remove the print 🙃

File: addons/ActionIcon/ActionIcon.gd [hunk 3]
```diff
@@ -313,14 +313,19 @@ func get_joypad_model(device: int) -> String:
...
+		if device_name.find("Xbox 360") > -1:
+			model = "Xbox360"
+		elif device_name.find("PS3") > -1:
+			model = "DS3"
+		elif device_name.find("PS4") > -1:
+			model = "DS4"
+		elif device_name.find("PS5") > -1:
+			model = "DualSense"
+		elif device_name.find("Joy-Con") > -1 or device_name.find("Joy Con") > -1:
+			model = "Joy_Con"
```
Comment (KoBeWi) [inline addons/ActionIcon/ActionIcon.gd:325]: You renamed Joy-Con to Joy_Con, which is less correct.

**Comment by KoBeWi** [comment]
Thanks for your work and sorry for waiting; I'm reviewing this now.

Some DS3 icons have wrong names (R1, R3). They should follow the button naming from XBox. Also Share button is missing for this set.

DS4 icons inlude Touchpad, which is unnecessary (unused).

DualSense set also includes extra unused icons: Microphone, Options/ShareAlt, Diagrams, TouchPad.

Xbox360 is missing Share and includes 2 extras: BackAlt and StartAlt.

All in all, ensure that all sets have exactly 28 buttons, they are the same buttons as in Xbox set and they have correct names.

File: addons/ActionIcon/ActionIcon.gd (hunk 2 above)
Comment (Bytezz) [reply addons/ActionIcon/ActionIcon.gd]: Heh heh, debug stuff... forgot to wipe it up.

**Comment by Bytezz** [comment]
> Thanks for your work and sorry for waiting; I'm reviewing this now.

Don't worry, these are just some edits I've done for some projects of mine.

> Some DS3 icons have wrong names (R1, R3).

Missed.

> They should follow the button naming from XBox. Also Share button is missing for this set.
> DS4 icons inlude Touchpad, which is unnecessary (unused). 
> DualSense set also includes extra unused icons: Microphone, Options/ShareAlt, Diagrams, TouchPad.
> Xbox360 is missing Share and includes 2 extras: BackAlt and StartAlt.

I've left these icons to enable devs to use them through CUSTOM_ACTIONS or some edits to the code.  
I don't think that's a problem to keep them because they are not such big.

File: addons/ActionIcon/ActionIcon.gd (hunk 3 above)
Comment (Bytezz) [reply addons/ActionIcon/ActionIcon.gd:325]: That's to follow JoypadModel enum's name (Joy_Con), which can't have a minus in the name "-".

**Comment by KoBeWi** [comment]
This still needs some work but I'll just merge it and clean-up with the Godot 4 upgrade.
Thanks for contributing.

**Pull request merged**